gtsam/nonlinear/tests/testValues.cpp

```diff
@@ -589,6 +589,23 @@ TEST(Values, MatrixDynamicInsertFixedRead) {
   CHECK_EXCEPTION(values.at<Matrix23>(key1), exception);
 }
 
+TEST(Values, Demangle) {
+  Values values;
+  Matrix13 v; v << 5.0, 6.0, 7.0;
+  values.insert(key1, v);
+  string expected = "Values with 1 values:\nValue v1: (Eigen::Matrix<double, 1, 3, 1, 1, 3>) [\n	5, 6, 7\n]\n\n";
+
+  stringstream buffer;
+  streambuf * old = cout.rdbuf(buffer.rdbuf());
```

Comment:
Great idea for a test!

I've been thinking for a long time to add an additional optional std::ostream parameter to all print() functions to simplify this kind of things. But yours is the kind of hack that is required as things are now :+1: